### ISSUE TITLE
Support cryptographic functions from AWS KMS

### ISSUE THREAD

**Issue by jtalir** [CZ-NIC/django-eidas-specific-node#86]
There should be a possibility to use https://aws.amazon.com/kms/ for cryptographic functions. One possible integration is described in the fork https://github.com/CZ-NIC/django-eidas-specific-node/compare/master...vrk-kpa:kms_support

**Comment by fenryxo** [comment]
@tpazderka, could you make a design review? Thanks.

**Comment by fenryxo** [comment]
Hello @samikarvonen, could you join us for the design review? Thanks.

**Comment by tpazderka** [comment]
I do believe that the `AwsKmsSigner` should be able to prepare the needed structure for the signature withou the use of `XmlSecSigner`.

I think that the default implementation of creating the structure could be in the ABC as a `_presign` method?

**Comment by samikarvonen** [comment]
Yes, it would be better not to sign the xml twice. I just didn't find any simple way of preparing the xml for signing in my implementation. Also, I think specifying that exact version of boto3 is unnecessary latest should work just fine.

**Comment by fenryxo** [comment]
> I do believe that the `AwsKmsSigner` should be able to prepare the needed structure for the signature withou the use of `XmlSecSigner`.

It isn't only about creating the structure but about filling the structure with certificate data and `<ds:SignedInfo>` data. AWS KMS client only signs the contents of `<ds:SignedInfo>` to fill `<ds:SignatureValue>`.
